include/exadg/operators/interior_penalty_parameter.h

```diff
@@ -44,48 +44,87 @@ calculate_penalty_parameter(AlignedVector<VectorizedArray<Number>> & array_penal
                             MatrixFree<dim, Number> const &          matrix_free,
                             unsigned int const                       dof_index = 0)
 {
-  unsigned int n_cells = matrix_free.n_cell_batches() + matrix_free.n_ghost_cell_batches();
+  unsigned int const n_cells = matrix_free.n_cell_batches() + matrix_free.n_ghost_cell_batches();
   array_penalty_parameter.resize(n_cells);
 
-  Mapping<dim> const &       mapping = *matrix_free.get_mapping_info().mapping;
-  FiniteElement<dim> const & fe      = matrix_free.get_dof_handler(dof_index).get_fe();
-  unsigned int const         degree  = fe.degree;
+  // Compute cell volume with usual FEEvaluation object and store it in
+  // array_penalty_parameter for now
+  {
+    FEEvaluation<dim, -1, 0, 1, Number> fe_eval(matrix_free, dof_index);
```

Comment:
Is it correct that this uses precompiled code for fast evaluation only up to degree 6 as written in the dealii docu? In ExaDG, we have the parameter `EXADG_DEGREE_MAX`, so why don't you use `CellIntegrator` and/or `FaceIntegrator` as in other parts of ExaDG?

Reply:
`FEEvaluation<dim, -1, 0, n_components, Number>` **is** `CellIntegrator<dim, n_components, Number`: https://github.com/exadg/exadg/blob/01939cbd3441629dacd624d5cef041cbb9904db6/include/exadg/matrix_free/integrators.h#L31-L35
So we get the same maximal degree in either case, relying on the order implied through the dynamic linking / shared library load (dlopen) to pick up ExaDG's maximal degree rather than deal.II's. It's right that we should switch to `CellIntegrator` here, it's just my habits picking the other :wink: